scripts/prohibit-java-lines-split-needlessly

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
+source "$self_dir"/utils.sh
+
+awk_script="$(cat <<'EOF'
+  function indent_width(str) {
+    for (i = 1; i <= length(str); i++) {
+      if (substr(str, i, 1) != " ") {
+        return i - 1;
+      }
+    }
+    return 0;
+  }
+
+  {
+    this_indent_width = indent_width($0);
+  }
+
+  prev ~ /,$/ && $0 ~ /[;{]$/ && prev_indent_width < this_indent_width {
```

Comment:
`$0` means "this line"; `<condition> { <statements> }` means "do statements for each line for which the condition applies"

Reply:
aka `file.lines().filter(condition).foreach($0 -> statements($0))